pkgs/by-name/sa/salmon/package.nix

```diff
@@ -0,0 +1,82 @@
+{ lib
+, stdenv
+, autoreconfHook
+, bash
+, boost
+, bzip2
+, cereal_1_3_2
+, cmake
+, curl
+, fetchFromGitHub
+, gcc
+, jemalloc
+, lbzip2
+, libgff
+, libiconv
+, libstaden-read
+, lzlib
+, pkg-config
+, tbb_2021_8
+, xz
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "salmon";
+  version = "1.10.2";
+
+  pufferFishSrc = fetchFromGitHub {
+      owner = "COMBINE-lab";
+      repo = "pufferfish";
+      rev = "salmon-v${version}";
+      sha256 = "sha256-JKbUFBEsqnENl4vFqve1FCd4TI3n9bRi2RNHC8QGQGc=";
+  };
+
+  src = fetchFromGitHub {
+          owner = "COMBINE-lab";
+          repo = "salmon";
+          rev = "v${version}";
+          sha256 = "sha256-kwqoUmVCqjr/xRxJjQKaFjjCQW+MFASHJ2f9OiAumNU=";
+        };
+
+  patches = [
+    # Use pufferfish source fetched by nix
+    ./fetch-pufferfish.patch
+  ];
+
+  postPatch = "patchShebangs .";
+
+  buildInputs = [
+    zlib
+    lzlib
+    lbzip2
+    xz
+    bzip2
+    cereal_1_3_2
+    tbb_2021_8
+    libgff
+    libstaden-read
+  ] ++ lib.optionals stdenv.isDarwin [ libiconv ];
+
+  nativeBuildInputs = [ cmake boost pkg-config bash curl jemalloc ];
```

Comment:
```suggestion
    libstaden-read
    jemalloc
    boost
    curl
  ] ++ lib.optionals stdenv.isDarwin [ libiconv ];

  nativeBuildInputs = [ cmake pkg-config ];

  strictDeps = true;
```

Reply:
So you moved some stuff from native to regular build inputs, and then set strictDeps to true? What is this sorcery? Do you have any recommended reading for me?

Reply:
> What is this sorcery? Do you have any recommended reading for me?

The difference between nativeBuildInputs and buildInputs is that the first set of packages is used at build time (when compiling) whereas the second set is required by the program that is being compiled. Enabling `strictDeps` enforces this separation a bit more than the default (lenient) settings. This is explained in more detail in the [docs](https://ryantm.github.io/nixpkgs/stdenv/stdenv/#ssec-stdenv-dependencies-overview).